internal/go_repository.bzl

```diff
@@ -188,8 +190,11 @@ def _go_repository_impl(ctx):
         for key in ("urls", "strip_prefix", "type", "sha256", "commit", "tag", "vcs", "remote"):
             if getattr(ctx.attr, key):
                 fail("cannot specify both version and %s" % key)
-        if ctx.attr.version and not ctx.attr.sum:
-            fail("if version is specified, sum must also be")
+        if not ctx.attr.sum:
+            if is_module_extension_repo:
+                fail("No sum for {}@{} found, run bazel run @rules_go//go -- mod tidy to generate it".format(ctx.attr.importpath, ctx.attr.version))
+            else:
+                fail("if version is specified, sum must also be")
```

Comment:
This required two more fixes: Replacing `print` with `fail` and also showing the new error message only with Bzlmod.

Reply:
@Buzz-Lightyear No need to take action, I applied the fixes and will merge now.